### ISSUE TITLE
Custom metrics: generate custom metrics values

### ISSUE THREAD

**Issue by mattab** [piwik/piwik#12390]
The [Custom Reports feature](https://piwik.org/docs/custom-reports/) is good at mixing dimensions together (up to 3 levels of dimensions), but all the metrics available are mostly "Standard metrics" in Piwik and a custom report can only contain one of the 100+ metrics available by default in Piwik.

Custom Metrics are especially useful to process a "KPI" value for a given day or week or month.  From user feedback:

> An example of such a metric would be the ratio between successful vs. total searches or the monthly/weekly/daily active user count. We define active as done at least 1 search per week (a metric that could also change in the future).

Custom Metrics would let us create new custom metrics and generate a widget / report showing the value of the metric for the selected website and time period.

**Comment by mattab** [comment]
Ideally this would be into core, or in Custom Dimensions, but this may need to be implemented in the Custom Reports plugin if not possible otherwise?

**Comment by tsteur** [comment]
We could move it to 4.1

**Comment by aleksijohansson** [comment]
Any updates on this?

**Comment by Jaimeapo** [comment]
Just another reminder to schedule this feature in a closer release??

**Comment by AkshayRao27** [comment]
Agreed. Not being able to calculate an an interaction rate for events that I've defined is quite frustrating. Defining Goals or Content Items gets me somewhat close, but it's still not flexible enough.
